UnityProject/Assets/Scripts/Systems/Antagonists/Antags/Blob/BlobPlayer.cs

```diff
@@ -0,0 +1,1275 @@
+						"Alert! Station integrity near critical. Biomass sensor levels are off the charts."),
+					MatrixManager.MainStationMatrix);
+				SoundManager.PlayNetworked("Notice1");
+			}
+
+			//Blob wins after number of blob tiles reached
+			if (!victory && numOfBlobTiles >= numOfTilesForVictory)
+			{
+				victory = true;
+				BlobWins();
+			}
+
+			//Detection check
+			if (!announcedBlob && (timer >= announceAfterSeconds || numOfBlobTiles >= numOfTilesForDetection))
+			{
+				announcedBlob = true;
+
+				Chat.AddSystemMsgToChat(
+					string.Format(CentComm.BioHazardReportTemplate,
+						"Confirmed outbreak of level 5 biohazard aboard the station. All personnel must contain the outbreak."),
+					MatrixManager.MainStationMatrix);
+				SoundManager.PlayNetworked("Outbreak5");
+			}
+
+			BiomassTick();
+
+			TrySpawnSpores();
+
+			AutoExpandBlob();
+
+			HealthPulse();
+
+			if (blobCore == null) return;
+
+			health = coreHealth.integrity;
+		}
+
+		[TargetRpc]
+		private void TargetRpcTurnOnClientLight(NetworkConnection target)
+		{
+			overmindLightObject.SetActive(true);
+			overmindLight.Color = color;
+			overmindLight.Color.a = 0.2f;
+		}
+
+		#region teleport
+
+		private IEnumerator TeleportPlayerBack()
+		{
+			Chat.AddExamineMsgFromServer(gameObject,
+				"You feel lost without blob.<color=#FF151F>Move back to the blob</color>");
+
+			yield return WaitFor.Seconds(1f);
+
+			if (ValidateAction(playerSync.ServerPosition, true))
+			{
+				teleportCheck = false;
+				yield break;
+			}
+
+			Chat.AddExamineMsgFromServer(gameObject, "Your mind gets sucked back to the blob");
+
+			if(blobCore == null) yield break;
+
+			TeleportToNode();
+
+			teleportCheck = false;
+		}
+
+		[Command]
+		public void CmdTeleportToCore()
+		{
+			if(blobCore == null) return;
+
+			playerSync.SetPosition(blobCore.WorldPosServer());
+		}
+
+		[Command]
+		public void CmdTeleportToNode()
+		{
+			TeleportToNode();
+		}
+
+		private void TeleportToNode()
+		{
+			GameObject node = null;
+
+			var vector = float.PositiveInfinity;
+
+			var pos = playerSync.ServerPosition;
+
+			//Find closet node
+			foreach (var blobStructure in nodeBlobs)
+			{
+				var distance = Vector3.Distance(blobStructure.location, pos);
+
+				if (distance > vector) continue;
+
+				vector = distance;
+				node = blobStructure.gameObject;
+			}
+
+			//If null go to core instead
+			if (node == null)
+			{
+				//Blob is dead :(
+				if(blobCore == null) return;
+
+				playerSync.SetPosition(blobCore.WorldPosServer());
+				return;
+			}
+
+			playerSync.SetPosition(node.WorldPosServer());
+		}
+
+		#endregion
+
+		#region SyncVars
+
+		private void SyncResources(int oldVar, int newVar)
+		{
+			resources = newVar;
+			resourceText.text = newVar.ToString();
+		}
+
+		private void SyncHealth(float oldVar, float newVar)
+		{
+			health = newVar;
+			healthText.text = newVar.ToString();
+		}
+
+		private void SyncNumOfBlobTiles(int oldVar, int newVar)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldVar'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6357029)